File: src/main/java/ru/timeconqueror/timecore/common/capability/property/serializer/StringPropertySerializer.java
package ru.timeconqueror.timecore.common.capability.property.serializer;

import net.minecraft.nbt.CompoundTag;

public class StringPropertySerializer implements IPropertySerializer<String> {
    public static final IPropertySerializer<String> INSTANCE = new StringPropertySerializer();
    public static final IPropertySerializer<String> NULLABLE = IPropertySerializer.nullable(INSTANCE);

    @Override
    public void serialize(String name, String value, CompoundTag nbt) {
        nbt.putString(name, value);
    }

    @Override
    public String deserialize(String name, CompoundTag nbt) {
        return nbt.getString(name);
    }
}

File: src/main/java/ru/timeconqueror/timecore/common/capability/property/serializer/StringPropertySerializer.kt
package ru.timeconqueror.timecore.common.capability.property.serializer

import net.minecraft.nbt.CompoundTag


object StringPropertySerializer : IPropertySerializer<String> {

    override fun serialize(name: String, value: String, nbt: CompoundTag) = nbt.putString(name, value)
    override fun deserialize(name: String, nbt: CompoundTag) = nbt.getString(name)

    object Nullable : NullPropertySerializer<String>(this)
}